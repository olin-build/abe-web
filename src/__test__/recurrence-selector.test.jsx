import moment from 'moment';
import React from 'react';
import renderer from 'react-test-renderer';
import RecurrenceSelector from '../pages/add-edit/recurrence-selector';

describe('RecurrenceSelector', () => {
  test('matches daily snapshot', () => {
    const recurrence = { count: 2, frequency: 'DAILY', interval: 1 };
    const jsx = (
      <RecurrenceSelector recurs={recurrence} start={moment('2018-05-12')} onChange={() => {}} />
    );
    const component = renderer.create(jsx);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('matches weekly snapshot', () => {
    const recurrence = {
      by_day: ['MO'],
      count: 2,
      frequency: 'WEEKLY',
      interval: 1,
    };
    const jsx = (
      <RecurrenceSelector recurs={recurrence} start={moment('2018-05-12')} onChange={() => {}} />
    );
    const component = renderer.create(jsx);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('matches montly snapshot', () => {
    const recurrence = {
      by_month_day: true,
      count: 2,
      frequency: 'MONTHLY',
      interval: 1,
    };
    const jsx = (
      <RecurrenceSelector recurs={recurrence} start={moment('2018-05-12')} onChange={() => {}} />
    );
    const component = renderer.create(jsx);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
