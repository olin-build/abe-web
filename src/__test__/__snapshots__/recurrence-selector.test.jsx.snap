// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecurrenceSelector matches daily snapshot 1`] = `
<div
  className="recurrence-container"
>
  <div
    className="radio-collection-container"
  >
    <span
      className="radio-collection-title"
    >
      Repeat
    </span>
    <div
      className="radio-collection-options-container"
    >
      <div
        className="radio-option"
      >
        <input
          checked={false}
          id="frequency-weekly"
          name="frequency"
          onChange={[Function]}
          title="Weekly"
          type="radio"
          value="WEEKLY"
        />
        <label
          htmlFor="frequency-weekly"
        >
          Weekly
        </label>
      </div>
      <div
        className="radio-option"
      >
        <input
          checked={false}
          id="frequency-monthly"
          name="frequency"
          onChange={[Function]}
          title="Monthly"
          type="radio"
          value="MONTHLY"
        />
        <label
          htmlFor="frequency-monthly"
        >
          Monthly
        </label>
      </div>
      <div
        className="radio-option"
      >
        <input
          checked={false}
          id="frequency-yearly"
          name="frequency"
          onChange={[Function]}
          title="Yearly"
          type="radio"
          value="YEARLY"
        />
        <label
          htmlFor="frequency-yearly"
        >
          Yearly
        </label>
      </div>
    </div>
  </div>
  <span
    className="radio-collection-title"
  >
    every
  </span>
  <input
    className="single-line-text-box super-short-text-box"
    min="1"
    onChange={[Function]}
    placeholder="#"
    title="Interval"
    type="number"
    value={1}
  />
  <div
    className="radio-collection-container"
  >
    <span
      className="radio-collection-title"
    >
      ends
    </span>
    <div
      className="radio-collection-options-container"
    >
      <div
        className="radio-option"
      >
        <input
          checked={false}
          id="end-options-date"
          name="end-options"
          onChange={[Function]}
          title="date"
          type="radio"
          value="date"
        />
        <label
          htmlFor="end-options-date"
        >
          date
        </label>
      </div>
      <div
        className="radio-option"
      >
        <input
          checked={false}
          id="end-options-occurences"
          name="end-options"
          onChange={[Function]}
          title="occurences"
          type="radio"
          value="occurences"
        />
        <label
          htmlFor="end-options-occurences"
        >
          <input
            className="single-line-text-box super-short-text-box"
            min="1"
            onChange={[Function]}
            placeholder="#"
            title="Occurence Number"
            type="number"
            value="2"
          />
          <span>
             times
          </span>
        </label>
      </div>
      <div
        className="radio-option"
      >
        <input
          checked={false}
          id="end-options-forever"
          name="end-options"
          onChange={[Function]}
          title="forever"
          type="radio"
          value="forever"
        />
        <label
          htmlFor="end-options-forever"
        >
          forever
        </label>
      </div>
    </div>
  </div>
  <div
    className="large-collapse"
  >
    <span>
      Repeats 
    </span>
    daily 
    
     2 times
  </div>
</div>
`;

exports[`RecurrenceSelector matches montly snapshot 1`] = `
<div
  className="recurrence-container"
>
  <div
    className="radio-collection-container"
  >
    <span
      className="radio-collection-title"
    >
      Repeat
    </span>
    <div
      className="radio-collection-options-container"
    >
      <div
        className="radio-option"
      >
        <input
          checked={false}
          id="frequency-weekly"
          name="frequency"
          onChange={[Function]}
          title="Weekly"
          type="radio"
          value="WEEKLY"
        />
        <label
          htmlFor="frequency-weekly"
        >
          Weekly
        </label>
      </div>
      <div
        className="radio-option"
      >
        <input
          checked={true}
          id="frequency-monthly"
          name="frequency"
          onChange={[Function]}
          title="Monthly"
          type="radio"
          value="MONTHLY"
        />
        <label
          htmlFor="frequency-monthly"
        >
          Monthly
        </label>
      </div>
      <div
        className="radio-option"
      >
        <input
          checked={false}
          id="frequency-yearly"
          name="frequency"
          onChange={[Function]}
          title="Yearly"
          type="radio"
          value="YEARLY"
        />
        <label
          htmlFor="frequency-yearly"
        >
          Yearly
        </label>
      </div>
    </div>
  </div>
  <span
    className="radio-collection-title"
  >
    every
  </span>
  <input
    className="single-line-text-box super-short-text-box"
    min="1"
    onChange={[Function]}
    placeholder="#"
    title="Interval"
    type="number"
    value={1}
  />
  <div
    className="radio-collection-container"
  >
    <span
      className="radio-collection-title"
    >
      on
    </span>
    <div
      className="radio-collection-options-container"
    >
      <div
        className="radio-option"
      >
        <input
          checked={false}
          id="month-options-month"
          name="month-options"
          onChange={[Function]}
          title="day of the month"
          type="radio"
          value="month"
        />
        <label
          htmlFor="month-options-month"
        >
          day of the month
        </label>
      </div>
      <div
        className="radio-option"
      >
        <input
          checked={false}
          id="month-options-week"
          name="month-options"
          onChange={[Function]}
          title="day of the week"
          type="radio"
          value="week"
        />
        <label
          htmlFor="month-options-week"
        >
          day of the week
        </label>
      </div>
    </div>
  </div>
  <div
    className="radio-collection-container"
  >
    <span
      className="radio-collection-title"
    >
      ends
    </span>
    <div
      className="radio-collection-options-container"
    >
      <div
        className="radio-option"
      >
        <input
          checked={false}
          id="end-options-date"
          name="end-options"
          onChange={[Function]}
          title="date"
          type="radio"
          value="date"
        />
        <label
          htmlFor="end-options-date"
        >
          date
        </label>
      </div>
      <div
        className="radio-option"
      >
        <input
          checked={false}
          id="end-options-occurences"
          name="end-options"
          onChange={[Function]}
          title="occurences"
          type="radio"
          value="occurences"
        />
        <label
          htmlFor="end-options-occurences"
        >
          <input
            className="single-line-text-box super-short-text-box"
            min="1"
            onChange={[Function]}
            placeholder="#"
            title="Occurence Number"
            type="number"
            value="2"
          />
          <span>
             times
          </span>
        </label>
      </div>
      <div
        className="radio-option"
      >
        <input
          checked={false}
          id="end-options-forever"
          name="end-options"
          onChange={[Function]}
          title="forever"
          type="radio"
          value="forever"
        />
        <label
          htmlFor="end-options-forever"
        >
          forever
        </label>
      </div>
    </div>
  </div>
  <div
    className="large-collapse"
  >
    <span>
      Repeats 
    </span>
    monthly 
    on the 12th day of the month
     2 times
  </div>
</div>
`;

exports[`RecurrenceSelector matches weekly snapshot 1`] = `
<div
  className="recurrence-container"
>
  <div
    className="radio-collection-container"
  >
    <span
      className="radio-collection-title"
    >
      Repeat
    </span>
    <div
      className="radio-collection-options-container"
    >
      <div
        className="radio-option"
      >
        <input
          checked={true}
          id="frequency-weekly"
          name="frequency"
          onChange={[Function]}
          title="Weekly"
          type="radio"
          value="WEEKLY"
        />
        <label
          htmlFor="frequency-weekly"
        >
          Weekly
        </label>
      </div>
      <div
        className="radio-option"
      >
        <input
          checked={false}
          id="frequency-monthly"
          name="frequency"
          onChange={[Function]}
          title="Monthly"
          type="radio"
          value="MONTHLY"
        />
        <label
          htmlFor="frequency-monthly"
        >
          Monthly
        </label>
      </div>
      <div
        className="radio-option"
      >
        <input
          checked={false}
          id="frequency-yearly"
          name="frequency"
          onChange={[Function]}
          title="Yearly"
          type="radio"
          value="YEARLY"
        />
        <label
          htmlFor="frequency-yearly"
        >
          Yearly
        </label>
      </div>
    </div>
  </div>
  <span
    className="radio-collection-title"
  >
    every
  </span>
  <input
    className="single-line-text-box super-short-text-box"
    min="1"
    onChange={[Function]}
    placeholder="#"
    title="Interval"
    type="number"
    value={1}
  />
  <div
    className="radio-collection-container"
  >
    <span
      className="radio-collection-title"
    >
      on
    </span>
    <div
      className="radio-collection-options-container"
    >
      <div
        className="radio-option"
      >
        <input
          checked={false}
          id="week-options-SU"
          name="week-options"
          onChange={[Function]}
          title="Sun"
          type="checkbox"
          value="SU"
        />
        <label
          htmlFor="week-options-SU"
        >
          Sun
        </label>
      </div>
      <div
        className="radio-option"
      >
        <input
          checked={true}
          id="week-options-MO"
          name="week-options"
          onChange={[Function]}
          title="Mon"
          type="checkbox"
          value="MO"
        />
        <label
          htmlFor="week-options-MO"
        >
          Mon
        </label>
      </div>
      <div
        className="radio-option"
      >
        <input
          checked={false}
          id="week-options-TU"
          name="week-options"
          onChange={[Function]}
          title="Tue"
          type="checkbox"
          value="TU"
        />
        <label
          htmlFor="week-options-TU"
        >
          Tue
        </label>
      </div>
      <div
        className="radio-option"
      >
        <input
          checked={false}
          id="week-options-WE"
          name="week-options"
          onChange={[Function]}
          title="Wed"
          type="checkbox"
          value="WE"
        />
        <label
          htmlFor="week-options-WE"
        >
          Wed
        </label>
      </div>
      <div
        className="radio-option"
      >
        <input
          checked={false}
          id="week-options-TH"
          name="week-options"
          onChange={[Function]}
          title="Thu"
          type="checkbox"
          value="TH"
        />
        <label
          htmlFor="week-options-TH"
        >
          Thu
        </label>
      </div>
      <div
        className="radio-option"
      >
        <input
          checked={false}
          id="week-options-FR"
          name="week-options"
          onChange={[Function]}
          title="Fri"
          type="checkbox"
          value="FR"
        />
        <label
          htmlFor="week-options-FR"
        >
          Fri
        </label>
      </div>
      <div
        className="radio-option"
      >
        <input
          checked={false}
          id="week-options-SA"
          name="week-options"
          onChange={[Function]}
          title="Sat"
          type="checkbox"
          value="SA"
        />
        <label
          htmlFor="week-options-SA"
        >
          Sat
        </label>
      </div>
    </div>
  </div>
  <div
    className="radio-collection-container"
  >
    <span
      className="radio-collection-title"
    >
      ends
    </span>
    <div
      className="radio-collection-options-container"
    >
      <div
        className="radio-option"
      >
        <input
          checked={false}
          id="end-options-date"
          name="end-options"
          onChange={[Function]}
          title="date"
          type="radio"
          value="date"
        />
        <label
          htmlFor="end-options-date"
        >
          date
        </label>
      </div>
      <div
        className="radio-option"
      >
        <input
          checked={false}
          id="end-options-occurences"
          name="end-options"
          onChange={[Function]}
          title="occurences"
          type="radio"
          value="occurences"
        />
        <label
          htmlFor="end-options-occurences"
        >
          <input
            className="single-line-text-box super-short-text-box"
            min="1"
            onChange={[Function]}
            placeholder="#"
            title="Occurence Number"
            type="number"
            value="2"
          />
          <span>
             times
          </span>
        </label>
      </div>
      <div
        className="radio-option"
      >
        <input
          checked={false}
          id="end-options-forever"
          name="end-options"
          onChange={[Function]}
          title="forever"
          type="radio"
          value="forever"
        />
        <label
          htmlFor="end-options-forever"
        >
          forever
        </label>
      </div>
    </div>
  </div>
  <div
    className="large-collapse"
  >
    <span>
      Repeats 
    </span>
    weekly 
    on Mon
     2 times
  </div>
</div>
`;
